### ISSUE TITLE
(ptucker): This will break for dynamic shapes. 

### ISSUE THREAD

**Issue by piascikj** [imdone/tensorflow#1372]
sigmoid_cross_entropy_with_logits requires [batch_size, 1] labels.

----
*opened via [imdone.io](https://imdone.io) from a code comment on [53bdaa9233a14a62be616cfdd6e143b4deab5b6e](https://github.com/imdone/tensorflow/commit/53bdaa9233a14a62be616cfdd6e143b4deab5b6e) by A. Unique TensorFlower gardener@tensorflow.org*

----

https://github.com/imdone/tensorflow/blob/8cb6e535f6ea14b380aa86d425169adad682cb8c/tensorflow/contrib/learn/python/learn/estimators/head.py#L797-L803
